### PR TITLE
#2544 message fields preview

### LINKED ISSUE #2544: Message fields preview
Add an ability to add message key/contents fields for preview (via json paths, probably)
image

### PR DESCRIPTION
**Pull request by kristi-dev**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Fixes #2544
**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [x] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**
Screenshot 2022-10-17 at 15 18 30

Screenshot 2022-10-17 at 15 18 50
Screenshot 2022-10-17 at 15 19 22

### CODE REVIEW

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx [hunk 1]
```diff
@@ -60,6 +70,31 @@ const Message: React.FC<Props> = ({
 
   const [vEllipsisOpen, setVEllipsisOpen] = React.useState(false);
 
+  const renderFilteredJson = (
+    jsonValue?: string,
+    filters?: PreviewFilter[]
+  ) => {
+    if (!filters?.length || !jsonValue) return jsonValue;
+
+    let parsedJson: object = {};
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:79]:
```suggestion
    let parsedJson;
```

File: kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx [hunk 2]
```diff
@@ -60,6 +70,31 @@ const Message: React.FC<Props> = ({
...
+      parsedJson = JSON.parse(jsonValue);
+    } catch (e) {
+      // do nothing;
+    }
+
+    return (
+      <>
+        {filters.map((item) => (
+          <div>
+            {item.field}: {get(parsedJson, item.path)}
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Messages/Message.tsx:91]:
```suggestion
            {item.field}: {get(parsedJson || {}, item.path)}
```

**Review by workshur — COMMENTED** [review]
@Haarolean pls take a look 👀

**Comment by Haarolean** [comment]
> @Haarolean pls take a look 👀

already on it :)

**Comment by Haarolean** [comment]
Looks good, a few suggestions:
1. Let's make the created fields editable or display their paths at least
2. Let's display "x is required" right after user tries to press "ok". That's disturbing to see the red text instantly after you open up the modal window.

@kristi-dev

**Comment by kristi-dev** [comment]
> Looks good, a few suggestions:
> 
> 1. Let's make the created fields editable or display their paths at least
> 2. Let's display "x is required" right after user tries to press "ok". That's disturbing to see the red text instantly after you open up the modal window.
> 
> @kristi-dev

Done both 
https://somup.com/c36Yqov3Vc

**Review by Haarolean — APPROVED** [review]
@workshur hey, are you done with the review?

**Review by workshur — APPROVED** [review]

**Comment by Haarolean** [comment]
@armenuikafka I gave it a quick glance, works fine, please take a look, you might find something

**Review by armenuikafka — CHANGES_REQUESTED** [review]
could we keep the preview in case of cancelling the editing

https://user-images.githubusercontent.com/104780608/197735764-f3c55291-b512-4f30-b18c-8bc0e707188b.mov

**Comment by armenuikafka** [comment]
@Haarolean maybe we need some limitation for fields or we can show not all content

limitation

**Comment by Haarolean** [comment]
> @Haarolean maybe we need some limitation for fields or we can show not all content
> 
> limitation

@kristi-dev can we add some limit to show this? the rest can be hidden like this: "lorem impsum..."

**Comment by kristi-dev** [comment]
> > @Haarolean maybe we need some limitation for fields or we can show not all content
> > limitation
> 
> @kristi-dev can we add some limit to show this? the rest can be hidden like this: "lorem impsum..."

yes, fix this

**Comment by kristi-dev** [comment]
> could we keep the preview in case of cancelling the editing
> 
>  preview.edit.mov

i changed edit functionality, so now we can see the list of fields above while editing and also, we are able to just close without editing and that field remains for preview

**Review by armenuikafka — APPROVED** [review]

**Pull request merged**